dmriprep/interfaces/vectors.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""@authors: Derek Pisner, Oscar Estaban, Ariel Rokem"""
+from __future__ import division, print_function
+from nipype.interfaces.base import BaseInterface, BaseInterfaceInputSpec, TraitedSpec, File, traits, Directory
+
+
+class CheckVecsInputSpec(BaseInterfaceInputSpec):
+    """Input interface wrapper for CheckVecs"""
+    fbval = File(exists=True, mandatory=True)
+    fbvec = File(exists=True, mandatory=True)
+    sesdir = Directory(exists=True, mandatory=True)
+    dwi_file = File(exists=True, mandatory=True)
+    B0_NORM_EPSILON = traits.Float(50)
+    bval_scaling = traits.Bool(True)
+
+
+class CheckVecsOutputSpec(TraitedSpec):
+    """Output interface wrapper for CheckVecs"""
+    from dipy.core.gradients import GradientTable
+    gtab = traits.Instance(GradientTable)
+    is_hemi = traits.Bool()
+    pole = traits.Array()
+    b0_ixs = traits.Array()
+    gtab_tsv_file = File(exists=True, mandatory=True)
+    slm = traits.Str()
+
+
+class CheckVecs(SimpleInterface):
+    """Interface wrapper for CheckVecs"""
+    input_spec = CheckVecsInputSpec
+    output_spec = CheckVecsOutputSpec
+
+    def _run_interface(self, runtime):
+        from dmriprep.utils.vectors import checkvecs
+        [gtab, is_hemi, pole, b0_ixs, gtab_tsv_file, slm] = checkvecs(
+            self.inputs.fbval,
+            self.inputs.fbvec,
+            self.inputs.sesdir,
+            self.inputs.dwi_file,
+            self.inputs.B0_NORM_EPSILON,
+            self.inputs.bval_scaling)
+        setattr(self, '_gtab', gtab)
+        setattr(self, '_is_hemi', is_hemi)
+        setattr(self, '_pole', pole)
+        setattr(self, '_b0_ixs', b0_ixs)
+        setattr(self, '_gtab_tsv_file', gtab_tsv_file)
```

Comment:
I meant all of these can be replaced by:

```suggestion
self._results['gtab'] = gtab
self._results['is_hemi'] = is_hemi
self._results['pole'] = pole
self._results['b0_ixs'] = b0_ixs
self._results['gtab_tsv_file'] = gtab_tsv_file
```
and then the whole `list_outputs` method can be deleted.

Reply:
Got it!